src/Npgsql/NpgsqlStream.cs

```diff
@@ -0,0 +1,329 @@
+            => throw new NotImplementedException();
+
+        #endregion
+
+        #region Seek
+
+        /// <inheritdoc />
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            ValidateArguments(origin);
+            CheckCanSeek();
+            return SeekAsync(offset, origin, default, false).GetAwaiter().GetResult();
+        }
+
+        /// <inheritdoc />
+        public virtual Task<long> SeekAsync(long offset, SeekOrigin origin, CancellationToken cancellationToken)
+        {
+            ValidateArguments(origin);
+            CheckCanSeek();
+            if (cancellationToken.IsCancellationRequested)
+                return Task.FromCanceled<long>(cancellationToken);
+            using (NoSynchronizationContextScope.Enter())
+                return SeekAsync(offset, origin, cancellationToken, true);
+        }
+
+        /// <inheritdoc />
+        internal protected virtual Task<long> SeekAsync(long offset, SeekOrigin origin, CancellationToken cancellationToken, bool async)
+            => throw new NotImplementedException();
+
+        #endregion
+
+        #region Dispose
+
+        /// <inheritdoc />
+        internal protected bool IsDisposed { get; protected set; }
+
+        /// <inheritdoc />
+        internal protected virtual void CheckDisposed()
+        {
+            if (IsDisposed)
+                throw new ObjectDisposedException(GetType().FullName, "Object disposed");
+        }
+
+        /// <inheritdoc />
+        protected sealed override void Dispose(bool disposing)
+        {
+            if (IsDisposed || !disposing)
+                return;
+
+            DisposeAsync(false).GetAwaiter().GetResult();
+
+            IsDisposed = true;
+        }
+
+        /// <inheritdoc />
+#if !NET461 && !NETSTANDARD2_0
+        public override ValueTask DisposeAsync()
+#else
+        public ValueTask DisposeAsync()
+#endif
+        {
+            if (IsDisposed)
+                return default;
+
+            using (NoSynchronizationContextScope.Enter())
+                return DisposeAsync(true).Then(() =>
+                    IsDisposed = true
+                );
+        }
+
+        /// <inheritdoc />
+        internal protected virtual ValueTask DisposeAsync(bool async) => default;
+
+        #endregion
+
+        #region APM
+
+        /// <inheritdoc />
+        public sealed override IAsyncResult BeginRead(byte[] buffer, int offset, int count, AsyncCallback callback, object? state)
+            => ReadAsync(buffer, offset, count).AsApm(callback, state);
+
+        /// <inheritdoc />
+        public sealed override IAsyncResult BeginWrite(byte[] buffer, int offset, int count, AsyncCallback callback, object? state)
+            => WriteAsync(buffer, offset, count).AsApm(callback, state);
+
+        /// <inheritdoc />
+        public sealed override int EndRead(IAsyncResult asyncResult)
+            => ((Task<int>)asyncResult).GetAwaiter().GetResult();
```

Comment:
If `asyncResult` is null, there are nothing prevents you from getting NRE.

Reply:
Check added.